src/utils.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const squaredDistance = require('ml-distance-euclidean').squared;
+
+/**
+ * Calculates the sum of squared errors
+ * @ignore
+ * @param {Array<Array<Number>>} data - the [x,y,z,...] points to cluster
+ * @param {Array<Array<Number>>} centers - the K centers in format [x,y,z,...]
+ * @param {Array<Number>} clusterID - the cluster identifier for each data dot
+ * @returns {Number} the sum of squared errors
+ */
+function computeSSE(data, centers, clusterID) {
+    let sse = 0;
+    let nData = data.length;
+    let c = 0;
+    for (let i = 0; i < nData; i++) {
```

Comment:
Mutation of `let` bindings in loops is still very slow in V8. Can you please only use `var` or `const` in hot code?